nixos/modules/services/web-apps/bookstack.nix

```diff
@@ -0,0 +1,352 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.bookstack;
+  bookstack = cfg.package;
+  db = cfg.database;
+  mail = cfg.mail;
+
+  user = cfg.user;
+  group = config.services.nginx.group;
+
+  # shell script for local administration
+  artisan = pkgs.writeScriptBin "bookstack" ''
+    #! ${pkgs.runtimeShell}
+    cd ${bookstack}
+    sudo=exec
+    if [[ "$USER" != bookstack ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u bookstack'
+    fi
+    $sudo ${pkgs.php}/bin/php artisan $*
+  '';
+
+
+in {
+  options.services.bookstack = {
+
+    enable = mkEnableOption ''
+      Enable BookStack.
+      The default admin details are "admin@admin.com" with a password of "password". You should change these details immediately after logging in for the first time.
+      This will also enable <option>services.phpfpm</option> and <option>services.nginx</option>.
+      Local artisan commands are run by using the command <literal>bookstack</literal>, e.g. <code>bookstack cache:clear</code>.
+    '';
+
+    package = mkOption {
```

Comment:
Does this option add more value or confusion? If we provide a `dataDir` option and a `package` option this is redundant.

Is it easy enough to use a different version of the package via overlays?

Reply:
I've seen it done this way in many modules (e.g. dokuwiki), so I decided to include it. I guess it's good to leave it for users who would like to "pin" a version more easily.

Reply:
This is what overlays are used for. Either solution works. Whatever you think is more understandable to the user 👍

Reply:
I've thought about it and you're right, without the package option is a bit more simple. I've removed it.